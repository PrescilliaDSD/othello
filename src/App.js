import React from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'

import Game from './Game/index'

const App = () => {
  return (
    <Router>
      <Route exact path="/game">
        <Game />
      </Route>
    </Router>
  )
}

export default App
