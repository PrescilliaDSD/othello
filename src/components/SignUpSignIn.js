import React, { useState } from 'react'
import PropTypes from 'prop-types'

const SignUpSignIn = ({ setLoggedIn, message, setMessage }) => {
  const [pseudo, setPseudo] = useState('')
  const [password, setPassword] = useState('')
  const [confirmPassword, setConfirmPassword] = useState('')

  const createUser = () => {
    if (confirmPassword !== password) {
      setMessage({
        text: 'Les mots de passe doivent Ãªtre identiques!',
        status: 'error'
      })
    } else {
      const user = {
        pseudo,
        password
      }
      fetch('http://localhost:8888/signup', {
        method: 'POST',
        headers: {
          'Content-type': 'application/json'
        },
        body: JSON.stringify(user)
      })
        .then((res) => res.json())
        .then((res) => {
          setMessage({
            text: res.msg,
            status: 'success'
          })
          setPseudo('')
          setPassword('')
          setConfirmPassword('')
        })
    }
  }

  const login = () => {
    const user = {
      pseudo,
      password
    }
    fetch('http://localhost:8888/signin', {
      method: 'POST',
      headers: {
        'Content-type': 'application/json'
      },
      body: JSON.stringify(user)
    })
      .then((res) => res.json())
      .then((res) => {
        setMessage({
          text: res.msg,
          status: res.login ? 'success' : 'error'
        })
        setLoggedIn(res.login)
        sessionStorage.setItem('pseudo', pseudo)
        sessionStorage.setItem('password', password)
        setPseudo('')
        setPassword('')
      })
  }

  return (
    <main>
      <h1 className="welcome-title">Bienvenue sur ce jeu Othello !</h1>
      <p className={`message message-${message.status}`}>{message.text}</p>
      <div className="sign-forms">
        <section>
          <h2>S&apos;inscrire</h2>
          <form>
            <label htmlFor="signUpPseudo">Pseudo : <input type="text" id="signUpPseudo" onChange={(event) => { setPseudo(event.target.value) }} value={pseudo} /></label>
            <label htmlFor="signUpPassword">Mot de passe : <input type="password" id="signUpPassword" onChange={(event) => { setPassword(event.target.value) }} value={password} /></label>
            <label htmlFor="confirmPassword">Confirmer le mot de passe : <input type="password" id="confirmPassword" onChange={(event) => { setConfirmPassword(event.target.value) }} value={confirmPassword} /></label>
            <button type="button" onClick={createUser}>S&apos;inscrire</button>
          </form>
        </section>
        <section>
          <h2>Se connecter</h2>
          <form>
            <label htmlFor="signInPseudo">Pseudo : <input type="text" id="signInPseudo" onChange={(event) => { setPseudo(event.target.value) }} value={pseudo} /></label>
            <label htmlFor="signInPassword">Mot de passe : <input type="password" id="signInPassword" onChange={(event) => { setPassword(event.target.value) }} value={password} /></label>
            <button type="button" onClick={login}>Se connecter</button>
          </form>
        </section>
      </div>
    </main>
  )
}

SignUpSignIn.propTypes = {
  setLoggedIn: PropTypes.func.isRequired,
  message: PropTypes.object.isRequired,
  setMessage: PropTypes.func.isRequired
}

export default SignUpSignIn
