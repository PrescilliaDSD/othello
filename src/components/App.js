import React, { useState, useEffect } from 'react'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'

import Home from './Home'
import Game from './Game'
import SignUpSignIn from './SignUpSignIn'
import NotFound from './NotFound'

const App = () => {
  const [loggedIn, setLoggedIn] = useState(false)
  const [gameData, setGameData] = useState({})
  const [message, setMessage] = useState({
    text: '',
    status: ''
  })

  useEffect(() => {
    if (sessionStorage.getItem('pseudo') && sessionStorage.getItem('password')) {
      const user = {
        pseudo: sessionStorage.getItem('pseudo'),
        password: sessionStorage.getItem('password')
      }
      fetch('http://localhost:8888/signin', {
        method: 'POST',
        headers: {
          'Content-type': 'application/json'
        },
        body: JSON.stringify(user)
      })
        .then((res) => res.json())
        .then((res) => { setLoggedIn(res.login) })
    }
  })

  return (
    <Router>
      <Switch>
        { loggedIn &&
          <Route exact path="/">
            <Home setLoggedIn={setLoggedIn} gameData={gameData} setGameData={setGameData} message={message} setMessage={setMessage} />
          </Route>
        }
        { loggedIn &&
          <Route exact path="/game/:id" component={Game}>
            <Game gameData={gameData} setGameData={setGameData} />
          </Route>
        }
        { !loggedIn &&
          <Route>
            <SignUpSignIn setLoggedIn={setLoggedIn} message={message} setMessage={setMessage} />
          </Route>
        }
        <Route component={NotFound} />
      </Switch>
    </Router>
  )
}

export default App
