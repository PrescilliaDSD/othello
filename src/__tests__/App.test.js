import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { mount } from 'enzyme'
import { MemoryRouter } from 'react-router'
import Adapter from 'enzyme-adapter-react-16'

import App from '../components/App'
import SignUpSignIn from '../components/SignUpSignIn'

Enzyme.configure({ adapter: new Adapter() })

describe('Root component', () => {
  it('should render correctly', () => {
    const component = renderer
      .create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render SignUpSignIn component by default', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/']}>
        <App/>
      </MemoryRouter>
    )
    expect(wrapper.find(SignUpSignIn)).toHaveLength(1)
  })
})
