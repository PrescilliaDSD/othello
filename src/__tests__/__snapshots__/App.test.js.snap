// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root component should render correctly 1`] = `
<main>
  <h1
    className="welcome-title"
  >
    Bienvenue sur ce jeu Othello !
  </h1>
  <p
    className="message message-"
  >
    
  </p>
  <div
    className="sign-forms"
  >
    <section>
      <h2>
        S'inscrire
      </h2>
      <form>
        <label
          htmlFor="signUpPseudo"
        >
          Pseudo : 
          <input
            id="signUpPseudo"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label
          htmlFor="signUpPassword"
        >
          Mot de passe : 
          <input
            id="signUpPassword"
            onChange={[Function]}
            type="password"
            value=""
          />
        </label>
        <label
          htmlFor="confirmPassword"
        >
          Confirmer le mot de passe : 
          <input
            id="confirmPassword"
            onChange={[Function]}
            type="password"
            value=""
          />
        </label>
        <button
          onClick={[Function]}
          type="button"
        >
          S'inscrire
        </button>
      </form>
    </section>
    <section>
      <h2>
        Se connecter
      </h2>
      <form>
        <label
          htmlFor="signInPseudo"
        >
          Pseudo : 
          <input
            id="signInPseudo"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label
          htmlFor="signInPassword"
        >
          Mot de passe : 
          <input
            id="signInPassword"
            onChange={[Function]}
            type="password"
            value=""
          />
        </label>
        <button
          onClick={[Function]}
          type="button"
        >
          Se connecter
        </button>
      </form>
    </section>
  </div>
</main>
`;
